### PR TITLE
Add logging to find pinch point

### PR DESCRIPTION
**Pull request by peterwilson**

### CODE REVIEW

File: lib/amqp/consumer.rb [hunk 1]
```diff
@@ -345,6 +345,7 @@ def handle_cancel_ok(cancel_ok)
 
       # Handle the delivery only if the consumer still exists.
       # The broker has been known to deliver a few messages after the consumer has been shut down.
+      AMQP::Session.logger.info("consumer #{consumer.class.to_s}, basic_deliver: #{basic_deliver}, metadata: #{metadata}")
```
Comment (ColinDKelley) [inline lib/amqp/consumer.rb:348]: Do these have good `to_s` methods? Or do you need `.inspect`?

File: lib/amqp/session.rb [hunk 2]
```diff
@@ -930,9 +930,15 @@ def receive_frame(frame)
         begin
           receive_frameset(@frames[frame.channel])
         ensure
-          # for channel.close, frame.channel will be nil. MK.
-          clear_frames_on(frame.channel) if @frames[frame.channel]
+          begin
+            clear_frames_on(frame.channel) if @frames[frame.channel] # for channel.close, frame.channel will be nil. MK.
+          rescue Exception => ex
```
Comment (ColinDKelley) [inline lib/amqp/session.rb:935]: Are you positive you want `Exception`? That will rescue script errors and other major problems. I was thinking an `ensure_safe` might work which doesn't rescue at that low a level.

Comment (peterwilson) [inline lib/amqp/session.rb:935]: `ensure_safe` does actually rescue `Exception`, it's a pretty intense save. I can lower this down to a `StandardError` though if you'd prefer.

File: lib/amqp/consumer.rb (hunk 1 above)
Comment (peterwilson) [inline lib/amqp/consumer.rb:348]: I'll update these to inspect. Good point!

File: lib/amqp/session.rb (hunk 2 above)
Comment (ColinDKelley) [inline lib/amqp/session.rb:935]: Really?  https://github.com/Invoca/exception_handling/blob/master/lib/exception_handling/methods.rb#L27-L34
Does Ringswitch have its own copy of `ensure_safe`?!

Comment (ColinDKelley) [inline lib/amqp/session.rb:935]: It's weird to have a typo rescued, so I'd probably back down to `StandardError`.

**Pull request merged**